### PR TITLE
#1796: "try again"! reminder for volunteers after no contact made

### LINKED ISSUE #1796: "try again"! reminder for volunteers after no contact made
Part of epic #1017, dependent on #1785, #1791, #1780, #1781, #1782, and #1783

AS a **volunteer**,
I WANT to receive an SMS notification reminding me to reach out to `case_contact_types` I have been previously unsuccessful in reaching,
SO THAT I remember to follow up with these important members of my `casa_case`'s team or with my `casa_case`.

### How it works now
Volunteers do not currently receive this reminder. They have to remember to follow up with unsuccessful contact attempts on their own.

### How it should work instead
IF two weeks have passed since a volunteer has logged a `case_contact` where `contact_made=false`,
AND has yet to since log a `case_contact` for the same `case_contact_type` where `contact_made=true`,
AND the volunteer is still actively assigned to the `casa_case` corresponding to that `case_contact`,
THEN the volunteer should receive an SMS notification reminding them to try contacting that `case_contact_type` again.

### Acceptance Criteria
- [x] When the above conditions are met, the volunteer should receive an SMS notifications.
- [x] The SMS should read something like this, depending on character restrictions: 
_It's been two weeks since you've tried reaching [`case_contact_type`]. Try again! [deep link to log a new `case_contact`]_
- [x] volunteer should not receive this notification the same day they receive #1793
- [x] volunteer should not receive this notification for the same `case_contact_type` more than once every 30 days

### PR DESCRIPTION
**Pull request by harsohailB**
### What github issue is this PR for, if any?

Resolves #1796 

### What changed, and why?

* Updated `user_case_contact_types_reminder` table to be `user_reminder_time` where each column now represents time a reminder was last sent (simpler this way)
* Created service and task for sending out no contact made reminders (tasks runs daily)
* Abstracted out `sms_reminder_service` to be used by other reminder services such as this `no_contact_reminder` and previously created `court_date_report_due_reminder`

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪

* Wrote tests for model, service, and task added

### Screenshots please :)

N/A

### CODE REVIEW

**Comment by harsohailB** [comment]
Apologies for the number of file changes. Had to rename a table and thus change references in a bunch of places. This should simplify keeping track of reminder times tho.

File: app/services/court_report_due_sms_reminder_service.rb [hunk 1]
```diff
@@ -1,35 +1,15 @@
...
-
-    def create_short_link
-      if BASE_URL.blank?
-        raise "BASE_URL environment variable not defined"
-      end
-
-      short_url_service = ShortUrlService.new
-      short_url_service.create_short_url(BASE_URL + GENERATE_CASE_COURT_REPORT_LINK)
-      short_url_service.short_url
+      "Your court report is due on #{report_due_date}. Generate a court report to complete & submit here: " + create_short_link(GENERATE_CASE_COURT_REPORT_LINK)
```
Comment (7riumph) [inline app/services/court_report_due_sms_reminder_service.rb:12]: We should use the 'SmsBodyHelper' for text message bodies so they are easily accessible (stored in one place)

Comment (harsohailB) [reply app/services/court_report_due_sms_reminder_service.rb]: Oh nice! Done!

**Review by compwron — APPROVED** [review]
hopefully the migrations go well in qa- should be fine in prod.

**Pull request merged**